oms/src/main/java/com/gfg/oms/client/Client.java

```diff
@@ -0,0 +1,116 @@
+package com.gfg.oms.client;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.MismatchedInputException;
+import com.gfg.oms.Config;
+import com.gfg.oms.Request;
+import com.gfg.oms.Response;
+import com.gfg.oms.consignment.request.Create;
+import com.gfg.oms.response.Error;
+import com.gfg.oms.consignment.response.CreateConsignmentResponse;
+import lombok.AllArgsConstructor;
+
+import javax.validation.constraints.NotNull;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+@AllArgsConstructor
+public class Client {
+    @NotNull
+    private Config config;
+
+    public CreateConsignmentResponse doCreateConsignmentRequest(Create createConsignmentRequest) throws JsonProcessingException {
```

Comment:
I guess, this method will raise some questions. Some points to remember:
- for each request type we will have different response type
- for each request type there are multiple responses (as far as I see, totally incompatible to each other)
Therefore I see no possibility of making it anyhow nicer but I'm totally open to any suggestions.